sklearn/preprocessing/_function_transformer.py

```diff
@@ -237,7 +224,16 @@ def fit(self, X, y=None):
         self : object
             FunctionTransformer class instance.
         """
-        X = self._check_input(X, reset=True)
+        X = validate_data(
+            self,
+            X,
+            reset=True,
+            accept_sparse=self.accept_sparse,
+            dtype="numeric",
+            ensure_all_finite="allow-nan",
+            ensure_2d=True,
```

Comment:
``ensure_2d`` is used in ``validate_data`` even if ``check_array`` is skipped. So we should condition it to the value of ``validate`` as well. This should fix the CI failures
```suggestion
            ensure_2d=self.validate,
```